src/java.base/share/classes/java/net/URLConnection.java

```diff
@@ -667,10 +673,12 @@ public long getHeaderFieldLong(String name, long Default) {
      */
     @SuppressWarnings("deprecation")
     public long getHeaderFieldDate(String name, long Default) {
-        String value = getHeaderField(name);
-        try {
-            return Date.parse(value);
-        } catch (Exception e) { }
+        final String value = getHeaderField(name);
+        if (value != null) {
+            try {
+                return Date.parse(value);
+            } catch (NumberFormatException e) { }
```

Comment:
(not a reviewer) I think NumberFormatException isn't sufficient here. `Date.parse` might throw an `IllegalArgumentException`.

Reply:
@SirYwell Thanks for spotting this. You are right: [Date.parse](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Date.html#parse(java.lang.String))
```
Attempts to interpret the string s as a representation of a date and time.
If the attempt is successful...
If the attempt fails, an IllegalArgumentException is thrown.
```
@jaikiran I suggest to use the original code for catching exceptions: `} catch (Exception e) { }`

Reply:
You are right @SirYwell. It was a copy/paste error on my part. Thank you for catching it. I have updated the PR to fix that part and catch `Exception` like previously.